src/main/java/com/example/textprocessingtool/DataManagement.java

```diff
@@ -67,6 +67,39 @@ public String replaceText(String pattern, String text, String replacement){
         }
     }
 
+    /**
+     * Find count int.
+     *
+     * @param pattern   the pattern
+     * @param inputText the input text
+     * @return the int
+     */
+    public int findCount(String pattern, String inputText) {
+        Matcher matcher = getMatcher(pattern, inputText);
+        int count = 0;
+
+        while (matcher.find()) {
+            count++;
+        }
+
+        return count;
+    }
+
+    /**
+     * Count words int.
+     *
+     * @param inputText the input text
+     * @return the int
+     */
+    public int countWords(String inputText) {
+        if (inputText == null || inputText.trim().isEmpty()) {
+            return 0;
+        }
```

Comment:
this is optional but if an `if` condition has a single line as a return value you can ignore `{}`.

eg.
```java
if(a.isPresent()) return a;

// or

if(a.isEmpty) log.info("something......")
else retun a;
```